File: prime_number_generator.rs
// Prime number generator using the Sieve of Eratosthenes algorithm.

fn main(){
   const LIMIT: usize = 1000000;
   let mut number_may_be_prime:[bool;LIMIT]=[true;LIMIT];
   number_may_be_prime[0]=false;
   number_may_be_prime[1]=false;
   for n in 2..LIMIT
   {
      if number_may_be_prime[n]==true
      {
         println!("{n}");
         for m in (n*2..LIMIT).step_by(n)
         {
            number_may_be_prime[m]=false;
         }
      }
   }
   println!("End of program!");
}

File: primenumbergenerator.lua
-- Prime number generator using the Sieve of Eratosthenes algorithm.

LIMIT=1000000
numberMayBePrime={}
for n = 0, LIMIT do
    numberMayBePrime[n]=true
end

numberMayBePrime[0]=false
numberMayBePrime[1]=false

for n = 0, LIMIT do
    if numberMayBePrime[n]==true then
        print(n)
        for m =n*2, LIMIT, n do
            numberMayBePrime[m]=false
        end
    end
end
print("End of program!")
